### PR TITLE
Token management

### PR DESCRIPTION
**Pull request by jeffbanks**
#1199 changes to support a more flexible set of inputs for token management with http-rest scenarios.

### CODE REVIEW

File: adapter-http/src/main/resources/activities/documents-api/http-docsapi-search-basic.yaml [hunk 1]
```diff
@@ -8,27 +8,20 @@ description: |
...
-    get-lt: run driver=http tags==name:main-get-lt cycles===TEMPLATE(read-cycles,TEMPLATE(docscount,10)) page-size=TEMPLATE(page-size,3) fields=TEMPLATE(fields,%5b%5d) threads=<<threads:auto>> errors=timer,warn
-    get-and: run driver=http tags==name:main-get-and cycles===TEMPLATE(read-cycles,TEMPLATE(docscount,10)) page-size=TEMPLATE(page-size,3) fields=TEMPLATE(fields,%5b%5d) threads=<<threads:auto>> errors=timer,warn
-    get-or: run driver=http tags==name:main-get-or cycles===TEMPLATE(read-cycles,TEMPLATE(docscount,10)) page-size=TEMPLATE(page-size,3) fields=TEMPLATE(fields,%5b%5d) threads=<<threads:auto>> errors=timer,warn
-    get-or-single-match: run driver=http tags==name:main-get-or-single-match cycles===TEMPLATE(read-cycles,TEMPLATE(docscount,10)) page-size=TEMPLATE(page-size,3) fields=TEMPLATE(fields,%5b%5d) threads=<<threads:auto>> errors=timer,warn
+  rampup-write: run driver=http tags==name:"rampup-put.*" cycles===TEMPLATE(docscount,10) docpadding=TEMPLATE(docpadding,0) match-ratio=TEMPLATE(match-ratio,0.01) threads=<<threads:auto>> errors=timer,warn
```
Comment (jshook) [inline adapter-http/src/main/resources/activities/documents-api/http-docsapi-search-basic.yaml:11]: I'd like to see these on `>-` line continuations to make them easier to read.

File: nosqlbench-build-docs
Comment (jshook) [inline nosqlbench-build-docs:1]: What is this submodule added for?

**Review by jshook — CHANGES_REQUESTED** [review]
One minor request, but a question about a new module.

@jshook updated for final review.  Also includes the version requirements for scenario files that were a bit behind.

File: adapter-http/src/main/resources/activities/documents-api/http-docsapi-search-basic.yaml (hunk 1 above)
Comment (jeffbanks) [reply adapter-http/src/main/resources/activities/documents-api/http-docsapi-search-basic.yaml:11]: Updated to include separate lines.

File: nosqlbench-build-docs
Comment (jeffbanks) [reply nosqlbench-build-docs:1]: Not sure where that came from, good catch.  Removing it.

File: adapter-http/src/main/resources/activities/baselines/http-rest.md [hunk 2]
```diff
@@ -16,13 +16,15 @@ Currently supporting Stargate, with plans to make more generic where REST tokens
...
-### Example
-* auto_gen_token: Discard(); StargateToken('http://localhost:8081/v1/auth');
+### Example using all defaults.
+token=Discard(); Token('<<auth_token:>>','<<uri:http://localhost:8081/v1/auth>>', '<<uid:cassandra>>', '<<pswd:cassandra>>');
 
-Here `Discard()` will ignore the long input value received (by default for the binding), then the StargateToken function automates a check for a cached Stargate token.  
-If not already discovered, a request to the local running Stargate service will be invoked.  Each subsequent cycles utilizes the cached value until a timeout occurs or the workload is restarted.
+Here `Discard()` will ignore the long input value received (by default for the binding), then the Token function automates a check for a cached tokens from the endpoint provided.  
+If not already discovered, a request to the local running token service will be invoked.  
+Each subsequent cycles utilizes the cached value until a timeout occurs (default 30m) or the workload is restarted.
```
Comment (jeffreyscarpenter) [inline adapter-http/src/main/resources/activities/baselines/http-rest.md:27]: Is this saying the token will be regenerated? that's not necessary unless you don't use the token for 15 minutes.

Comment (jeffbanks) [reply adapter-http/src/main/resources/activities/baselines/http-rest.md:27]: Currently, the binding function will retain the token it has been provided or the one it requests from the Stargate coordinator.  After it gets the token from the coordinator, it caches it for reuse without another request to the coordinator, checking the delta date-time between first cached date-tinme and current date-time.  If >30m it requests another token from the coordinator.

File: adapter-http/src/main/resources/activities/baselines/http-rest-keyvalue.yaml [hunk 3]
```diff
@@ -15,9 +15,9 @@ scenarios:
 
 bindings:
 
-  weighted_hosts: ToString(); WeightedStrings('<<restapi_host:stargate>>'); ToString();
+  weighted_hosts: ToString(); WeightedStrings('<<stargate_host:stargate>>'); ToString();
   request_id: ToHashedUUID(); ToString();
-  auto_gen_token: Discard(); StargateToken('http://localhost:8081/v1/auth'); ToString();
+  token: Discard(); Token('<<auth_token:>>','<<uri:http://localhost:8081/v1/auth>>', '<<uid:cassandra>>', '<<pswd:cassandra>>');
```
Comment (jeffreyscarpenter) [inline adapter-http/src/main/resources/activities/baselines/http-rest-keyvalue.yaml:20]: Nit: perhaps `auth_uri`, `auth_uid`, `auth_pswd` would be more descriptive variable names?

**Comment by jeffreyscarpenter** [comment]
> @jshook and @jeffreyscarpenter checking in to see if there is anything else desired in this PR before merge?

Apologies for the delay. I have built and tested this locally and it works great! I added one more comment - a suggestion about variable names but it is not a requirement.

**Review by jeffreyscarpenter — APPROVED** [review]
Looks great!

**Pull request merged**